pkgs/applications/networking/mumble/default.nix

```diff
@@ -72,66 +49,63 @@ let
       ++ lib.optional speechdSupport speechd
-      ++ lib.optional pulseSupport libpulseaudio;
+      ++ lib.optional pulseSupport libpulseaudio
+      ++ lib.optional pipewireSupport pipewire;
 
     configureFlags = [
-      "CONFIG+=no-server"
-    ];
+      "-D server=OFF"
+      "-D bundled-celt=ON"
+      "-D bundled-opus=OFF"
+      "-D bundled-speex=OFF"
+      "-D bundled-rnnoise=OFF"
+      "-D bundle-qt-translations=OFF"
+      "-D update=OFF"
+      "-D overlay-xcompile=OFF"
+      "-D oss=OFF"
+    ] ++ lib.optional (!speechdSupport) "-D speechd=OFF"
+      ++ lib.optional (!pulseSupport) "-D pulseaudio=OFF"
+      ++ lib.optional (!pipewireSupport) "-D pipewire=OFF"
+      ++ lib.optional jackSupport "-D alsa=OFF -D jackaudio=ON";
 
     NIX_CFLAGS_COMPILE = lib.optional speechdSupport "-I${speechd}/include/speech-dispatcher";
 
-    installPhase = ''
-      # bin stuff
-      install -Dm755 release/mumble $out/bin/mumble
-      install -Dm755 scripts/mumble-overlay $out/bin/mumble-overlay
-
-      # lib stuff
-      mkdir -p $out/lib/mumble
-      cp -P release/libmumble.so* $out/lib
-      cp -P release/libcelt* $out/lib/mumble
-      cp -P release/plugins/* $out/lib/mumble
-
-      # icons
-      install -Dm644 scripts/mumble.desktop $out/share/applications/mumble.desktop
-      install -Dm644 icons/mumble.svg $out/share/icons/hicolor/scalable/apps/mumble.svg
+    postFixup = ''
+      wrapProgram $out/bin/mumble \
+        --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath (lib.optional pulseSupport libpulseaudio ++ lib.optional pipewireSupport pipewire)}"
     '';
   } source;
 
   server = source: generic {
     type = "murmur";
 
-    postPatch = lib.optionalString iceSupport ''
-      grep -Rl '/usr/share/Ice' . | xargs sed -i 's,/usr/share/Ice/,${zeroc-ice.dev}/share/ice/,g'
-    '';
-
     configureFlags = [
-      "CONFIG+=no-client"
-    ] ++ lib.optional (!iceSupport) "CONFIG+=no-ice"
-      ++ lib.optional grpcSupport "CONFIG+=grpc";
+      "-D client=OFF"
+    ] ++ lib.optional (!iceSupport) "-D ice=OFF"
+      ++ lib.optionals iceSupport [
+        "-D Ice_HOME=${lib.getDev zeroc-ice};${lib.getLib zeroc-ice}"
+        "-D CMAKE_PREFIX_PATH=${lib.getDev zeroc-ice};${lib.getLib zeroc-ice}"
+        "-D Ice_SLICE_DIR=${lib.getDev zeroc-ice}/share/ice/slice"
+      ]
+      ++ lib.optional grpcSupport "-D grpc=ON";
 
     buildInputs = [ libcap ]
       ++ lib.optional iceSupport zeroc-ice
       ++ lib.optionals grpcSupport [ grpc which ];
-
-    installPhase = ''
-      # bin stuff
-      install -Dm755 release/murmurd $out/bin/murmurd
-    '';
   } source;
 
   source = rec {
-    version = "1.3.4";
+    version = "unstable-1.4.231";
 
     # Needs submodules
     src = fetchFromGitHub {
       owner = "mumble-voip";
       repo = "mumble";
-      rev = version;
-      sha256 = "sha256-njovShQpW0JNeeE8VugdmGzDk3fKG8/fcJoBgdyTZug=";
+      rev = "9e0e274d6a9d8a9919267e747d05d0500d150560";
```

Comment:
This still needs a comment explaining how this revision came to be such that it can be updated properly in the future.

Reply:
To be honest, not sure what to comment there "that it can be updated properly in the future". I decided for that revision just because 1.4.230 is missing a lot of things, which are part of 1.4.231. In other words, for me the release of 1.4.230 is incomplete.

So as long as nothing critical appears and we have don't have to act, the next update will use an official stable release as usual.

Reply:
I interpreted your earlier comment https://github.com/NixOS/nixpkgs/pull/161663#discussion_r814995030 to mean that you chose 9e0e274d6a9d8a9919267e747d05d0500d150560 because it corresponds to the build number 231 in the CI. But from this comment it seems that this was just a coincidence that the commit you picked actually lines up with the unstable 231 release.

In any case, however we got here, the commit does line up with CI build number 231, which I also expect to be used for future release numbers. So at least while we use an unstable release, this code here should have a comment explaining this relation. Something along the lines of

```
# Revision for the unstable release number, see https://mumble.info/build-number.json
# For stable releases use the corresponding git tag instead
rev = "9e0e274d6a9d8a9919267e747d05d0500d150560";
```

Reply:
> I interpreted your earlier comment [#161663 (comment)](https://github.com/NixOS/nixpkgs/pull/161663#discussion_r814995030) to mean that you chose 9e0e274d6a9d8a9919267e747d05d0500d150560 because it corresponds to the build number 231 in the CI. But from this comment it seems that this was just a coincidence that the commit you picked actually lines up with the unstable 231 release.

What I wrote in the comment you linked still applies. A build number points to a specific commit (e.g. 231 --> 9e0e274) and if the developers decide a specific build number is stable, then they create a new release on their website.

So I wondered how to include these changes without creating lots of mess. I didn't want to use 1.4.230 as version number and also use the commit 9e0e274, because the version doesn't correspond to that commit and thus this is misleading.

it isn't a coincidence that I used this specific commit or that the build number points to it. One of the Mumble developers kindly created a new build number for me, as I mentioned in the comment you linked. As you can see in the commit history, I also backported various patches in the last weeks https://github.com/mumble-voip/mumble/commits/1.4.x. So this was intended.

> In any case, no matter we got here, the commit does line up with CI build number 231, which I also expect to be used for future (unstable, but also stable) release numbers. So at least while we use an unstable release, this code here should have a comment explaining this relation. Something along the lines of
> 
> ```
> # Revision for the unstable release number, see https://mumble.info/build-number.json
> # For stable releases use the corresponding git tag instead
> rev = "9e0e274d6a9d8a9919267e747d05d0500d150560";
> ```

Agree, let's add this. I can do that in my other PR #140147.